modules/dbnd/src/dbnd/_core/task_run/task_run_sync_local.py

```diff
@@ -27,25 +33,46 @@ def __init__(self, task_run):
                 and not p_val.fs.local
             ):
                 # Target requires local access, it points to a remote path that must be synced-to from a local path
-                local_target = target(
-                    self.local_sync_root,
-                    os.path.basename(p_val.path),
-                    config=p_val.config,
+                local_target = self._sync_local_target(p_val)
+
+                self._sync_input_or_output(p_def, p_val, local_target)
+
+            elif (
+                isinstance(p_val, MultiTarget)
+                and p_val.config.require_local_access
+                and not p_val.fs.local
```

Comment:
we probably should check every target in multitarget for fs.local

Reply:
yes, my bad, I've refactored this logic